internal/controller/controllers/cluster_controller.go

```diff
@@ -58,15 +62,35 @@ func (r *ClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		if k8serrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		return ctrl.Result{}, errors.Wrap(err, "failed to get cluster data")
+		r.Log.Info("failed to get resource %s", req.NamespacedName)
+		return ctrl.Result{Requeue: true}, nil
 	}
 
 	// check if new cluster
 	if cluster.Status.ID == "" {
+		r.Log.Info("Creating a new cluster %s %s", cluster.Name, cluster.Namespace)
 		return r.createNewCluster(ctx, cluster)
 	}
 
-	return ctrl.Result{}, nil
+	c, err := r.Installer.GetClusterInternal(ctx, installer.GetClusterParams{
```

Comment:
It is, i can try to call for the API with empty UUID but it will fail or the db will return the first cluster it finds, rest infra prevent this state of empty UUID. 
We talked about using labels, but i'm not sure if there is a way for the us to prevent users changing them. 
Another solution (not in this PR) will be adding another API to the backend that will know how to return an object based on name+namespace, eventually backed will have this info anyway because we will need to get notifications from it, instead of pulling every 10 seconds.

Reply:
opened https://issues.redhat.com/browse/MGMT-3539

Reply:
Will update the PR when this ticket is resolved.